keras_retinanet/bin/train.py

```diff
@@ -273,14 +284,21 @@ def main(args=None):
     train_generator, validation_generator = create_generators(args)
 
     # create the model
-    if args.snapshot:
+    if args.snapshot is not None:
         print('Loading model, this may take a second...')
         model            = keras.models.load_model(args.snapshot, custom_objects=custom_objects)
         training_model   = model
         prediction_model = model
     else:
+        weights = None
```

Comment:
Isn't this the same as:
```python
weights = args.weights
if args.imagenet_weights:
    weights = download_imagenet(RESNET_BACKBONE)
```
?

Reply:
Hmm yeah, it is if you know that --weights and --imagenet-weights are mutually exclusive. I suppose that is a reasonable thing to assume here.

Reply:
Actually, it isn't. `args.imagenet_weights` is always true, except when `--no-weights` is specified. But it can be simplified to:
```
weights = args.weights
if weights is None and args.imagenet_weights:
    weights = download_imagenet(RESNET_BACKBONE)
```